include/sparrow/layout/mutable_array_base.hpp

```diff
@@ -251,47 +299,47 @@ namespace sparrow
      * @param last The iterator to the element following the last element to insert.
      * @return An iterator pointing to the first element inserted, or \c pos if <tt>first == last</tt>.
      */
-    template <class D>
-    template <typename InputIt>
-    auto mutable_array_base<D>::insert(const_iterator pos, InputIt first, InputIt last) -> iterator
-    {
-        SPARROW_ASSERT_TRUE(pos >= this->cbegin())
-        SPARROW_ASSERT_TRUE(pos <= this->cend());
-        SPARROW_ASSERT_TRUE(first <= last);
-        const difference_type distance = std::distance(this->cbegin(), pos);
-        const auto validity_range = std::ranges::subrange(first, last)
-                                    | std::views::transform(
-                                        [](const value_type& obj)
-                                        {
-                                            return obj.has_value();
-                                        }
-                                    );
-        auto& derived = this->derived_cast();
-        derived.insert_bitmap(
-            sparrow::next(this->bitmap_cbegin(), distance),
-            validity_range.begin(),
-            validity_range.end()
-        );
-
-        const auto value_range = std::ranges::subrange(first, last)
-                                 | std::views::transform(
-                                     [](const value_type& obj)
-                                     {
-                                         return obj.get();
-                                     }
-                                 );
-        derived.insert_values(
-            sparrow::next(derived.value_cbegin(), distance),
-            value_range.begin(),
-            value_range.end()
-        );
-        const difference_type count = std::distance(first, last);
-        // The following must be done after modifying the bitmap and values
-        this->get_arrow_proxy().set_length(this->size() + static_cast<size_t>(count));
-
-        derived.update();
-        return sparrow::next(begin(), distance);
-    }
+    // template <class D>
+    // template <mpl::iterator_of_type<typename mutable_array_base<D>::value_type> InputIt>
```

Comment:
This modification leads to error:
out-of-line definition of 'insert' does not match any declaration in 'mutable_array_base<D>